### PR TITLE
Add elements for v2 sample sheet

### PR DESCRIPTION
**Pull request by diitaz93**
## Description

As mentioned in https://github.com/Clinical-Genomics/project-planning/issues/436, the NovaSeqX sequencing files require a different version of sample sheet for demultiplexing. The specifications of the new sample sheet are detailed in the [Illumina webpage](https://support-docs.illumina.com/SHARE/SampleSheetv2/Content/SHARE/SampleSheetv2/Settings_fNV_mX.htm). 

The proposed solution involves including a parameter to the CLI command specifying the version of the sample sheet.

### Added

- The parameter `sheet_version` to all the functions needed, from the CLI functions to the `SampleSheetCreator` class.
- Required sections of the v2 sample sheet.

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b novaseq-pt1 -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by diitaz93** [comment]
Not ready but I could get some feedback 🙂  @henrikstranneheim @Vince-janv @seallard

File: cg/apps/demultiplex/sample_sheet/novaseq_sample_sheet.py [hunk 1]
```diff
@@ -88,22 +96,37 @@ def convert_sample_to_header_dict(
         sample_dict = sample.dict(by_alias=True)
         return [str(sample_dict[header]) for header in sample_sheet_headers]
 
+    def create_v2_sample_sheet_sections(self) -> List[str]:
+        """Create the header and read sections of a v2 sample sheet in a list."""
+        return [
+            SAMPLE_SHEET_HEADER,
+            SAMPLE_SHEET_HEADER_FILE_FORMAT_V2,
+            SAMPLE_SHEET_HEADER_INSTRUMENT_TYPE_NOVASEQX,  # HARD-WIRED VALUE
+            SAMPLE_SHEET_HEADER_INSTRUMENT_PLATFORM,  # HARD-WIRED VALUE
```
Comment (diitaz93) [inline cg/apps/demultiplex/sample_sheet/novaseq_sample_sheet.py:105]: As shown in the [Illumina documentation](url[)](https://support-docs.illumina.com/SHARE/SampleSheetv2/Content/SHARE/SampleSheetv2/Settings_fNV_mX.htm), these two fields are mandatory but I have not found a way to extract this info from the RunParameters

File: cg/constants/demultiplexing.py [hunk 2]
```diff
@@ -38,6 +38,13 @@ class BclConverter(StrEnum):
     ],
 }
 
+SAMPLE_SHEET_HEADER = "[Header]"
+SAMPLE_SHEET_HEADER_FILE_FORMAT_V2 = "FileFormatVersion,2"
+SAMPLE_SHEET_HEADER_INSTRUMENT_TYPE_NOVASEQX = "NovaSeqX"
+SAMPLE_SHEET_HEADER_INSTRUMENT_PLATFORM = "NovaSeqXSeries"
```
Comment (diitaz93) [inline cg/constants/demultiplexing.py:44]: See comment above

File: cg/apps/demultiplex/sample_sheet/index.py [hunk 3]
```diff
@@ -37,9 +37,10 @@ class Index(BaseModel):
 
 
 def get_valid_indexes(dual_indexes_only: bool = True) -> List[Index]:
-    LOG.info("Fetch valid indexes from %s", valid_indexes_path)
+    """Return list of indexes from the valid indexes' path."""
+    LOG.info("Fetch valid indexes from %s", VALID_INDEXES_PATH)
     indexes: List[Index] = []
-    with open(valid_indexes_path, "r") as csv_file:
+    with open(VALID_INDEXES_PATH, "r") as csv_file:
```
Comment (henrikstranneheim) [inline cg/apps/demultiplex/sample_sheet/index.py:43]: Use new Io CVS module to read file

File: cg/apps/demultiplex/sample_sheet/novaseq_sample_sheet.py [hunk 4]
```diff
@@ -88,22 +96,37 @@ def convert_sample_to_header_dict(
         sample_dict = sample.dict(by_alias=True)
         return [str(sample_dict[header]) for header in sample_sheet_headers]
 
+    def create_v2_sample_sheet_sections(self) -> List[str]:
+        """Create the header and read sections of a v2 sample sheet in a list."""
+        return [
```
Comment (henrikstranneheim) [inline cg/apps/demultiplex/sample_sheet/novaseq_sample_sheet.py:101]: Should we create this via a pydantic module instead. We could create a base module for sheets and inherit from it for different type of sheets. Then we can also validate when creating

**Review by henrikstranneheim — COMMENTED** [review]
Good start 👍

**Pull request closed without merging**